### PR TITLE
feat: add KF_LANG env variable

### PR DESCRIPTION
**Pull request by wg102**
Resolves https://github.com/StatCan/jupyter-apis/issues/47

### CODE REVIEW

**Review by Ito-Matsuda — CHANGES_REQUESTED** [review]
Unsure if the following files also need to be modified to also have a field for language; [types.ts](https://github.com/StatCan/jupyter-apis/blob/feat-env-var/frontend/src/app/utils/types.ts) and [spawner_ui_config.yaml](https://github.com/StatCan/jupyter-apis/blob/feat-env-var/samples/spawner_ui_config.yaml). Only found those two files after searching with `shm` through this project.  The `spawner_ui_config` seems to set defaults and if it's read only, but the code works just fine without it (set elsewhere). As for `types.ts` I have a feeling it _might_ (make note of italics im not 100% sure) need to be in there somewhere.  I'm not super familiar with jupyter-apis but searching [import { Config } ](https://github.com/StatCan/jupyter-apis/search?q=import+%7B+Config+%7D) produces some results where the `Config` interface definition is used.

File: notebooks.go [hunk 1]
```diff
@@ -61,6 +62,7 @@ type newnotebookrequest struct {
 	DataVolumes        []volumerequest   `json:"datavols"`
 	EnableSharedMemory bool              `json:"shm"`
 	Configurations     []string          `json:"configurations"`
+	Language		   string			 `json:"language"`
```
[the remark this answers is not in the document]
Comment (wg102) [reply notebooks.go]: This was due to using tabs instead of spaces

File: notebooks.go [hunk 2]
```diff
@@ -471,6 +475,19 @@ func (s *server) NewNotebook(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	//Add Language
+	//Validate that the language format is valid (language[_territory])
+	match, err := regexp.MatchString("[[:alpha:]]{2}(_[[:alpha:]]{2})?$", req.Language)
```
Comment (brendangadd) [inline notebooks.go:480]: Matches anything that **ends** with a valid code. Needs a `^` in there.

File: notebooks.go [hunk 3]
```diff
@@ -23,6 +25,7 @@ import (
 const DefaultServiceAccountName string = "default-editor"
 const SharedMemoryVolumeName string = "dshm"
 const SharedMemoryVolumePath string = "/dev/shm"
+const EnvKFLanguage string = "KF_LANG"
```
Comment (brendangadd) [inline notebooks.go:28]: Treat acronyms in variable names as regular words for camelcasing.

File: notebooks.go [hunk 4]
```diff
@@ -471,6 +475,19 @@ func (s *server) NewNotebook(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	//Add Language
+	//Validate that the language format is valid (language[_territory])
+	match, err := regexp.MatchString("[[:alpha:]]{2}(_[[:alpha:]]{2})?$", req.Language)
+		if (err != nil || !match) {
+			var errLanguageFormat = errors.New("Error server language format")
```
Comment (brendangadd) [inline notebooks.go:482]: Make error message more clear, so someone reading logs in a console is likely to understand what occurred.

**Review by Ito-Matsuda — COMMENTED** [review]
Looks fine to me. Right now `shm` defaults to True/Enabled, but you mentioned that the change there needs to take place in the manifest.

File: frontend/src/assets/i18n/fr.json [hunk 5]
```diff
@@ -48,9 +48,11 @@
     "deleteDialogNo": "Annuler"
   },
   "formAdvancedOptions": {
-    "h3MiscSettings": "Paramètres divers",
-    "txtMiscSettings": "Autres paramètres pouvant être appliqué au serveur bloc-notes.",
-    "toggleSharedMemory": "Activer la mémoire partagée"
+    "h3AdvancedSettings": "Paramètres avancés",
+    "toggleSharedMemory": "Activer la mémoire partagée",
+    "lblSystemLanguage":"Language du système",
```
Comment (brendangadd) [inline frontend/src/assets/i18n/fr.json:53]: "Langue"?

**Review by brendangadd — CHANGES_REQUESTED** [review]
**Commit message:**
* Make title describe the feature rather than the variable
* Add a message body that provides some details on what you changed

**Comment by brendangadd** [comment]
@wg102 Looks great, and commit message is excellent. You can squash it down (keep the new commit message) and we'll merge.

**Pull request merged**